packages/design-system-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { forwardRef, useImperativeHandle, useState } from 'react';
+
+import { twMerge } from '../../utils/tw-merge';
+import { Icon, IconName, IconColor, IconSize } from '../Icon';
+import { Text } from '../Text';
+
+import type { CheckboxProps } from './Checkbox.types';
+
+export const Checkbox = forwardRef<{ toggle: () => void }, CheckboxProps>(
+  (
+    {
+      isSelected,
+      defaultIsSelected = false,
+      isDisabled = false,
+      isInvalid = false,
+      label,
+      labelProps,
+      onChange,
+      checkboxContainerProps,
+      checkedIconProps,
+      className,
+      style,
+      ...props
+    },
+    ref,
+  ) => {
+    const [internalSelected, setInternalSelected] = useState(defaultIsSelected);
+    const isControlled = isSelected !== undefined;
+    const currentSelected = isSelected ?? internalSelected;
+
+    const toggle = () => {
+      if (isDisabled) {
+        return;
+      }
+      const next = !currentSelected;
+      if (!isControlled) {
+        setInternalSelected(next);
+      }
+      onChange?.(next);
+    };
+
+    useImperativeHandle(ref, () => ({ toggle }), [toggle]);
+
+    const outerClassName = twMerge(
+      'inline-flex items-center',
+      isDisabled && 'cursor-not-allowed opacity-50',
+      className,
+    );
+
+    const baseBg = currentSelected
+      ? 'bg-primary-default hover:bg-primary-default-hover active:bg-primary-default-pressed'
+      : 'bg-background-default hover:bg-background-default-hover active:bg-background-default-pressed';
+    let baseBorder = 'border-border-default';
+    if (currentSelected) {
+      baseBorder = 'border-primary-default';
+    } else if (isInvalid) {
+      baseBorder = 'border-error-default';
+    }
+
+    const checkboxClasses = twMerge(
+      'flex h-[22px] w-[22px] items-center justify-center rounded border-2 transition-transform active:scale-95',
+      baseBg,
+      baseBorder,
+      checkboxContainerProps?.className,
+    );
+
+    const ariaLabel = typeof label === 'string' ? label : props['aria-label'];
+
+    return (
+      <button
```

Comment:
Have a look at the extension for correct semantic and simpler logic html https://github.com/MetaMask/metamask-extension/blob/main/ui/components/component-library/checkbox/checkbox.tsx